src/Symfony/Bundle/FrameworkBundle/Command/MessageConsumeCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Message\MessageBusInterface;
+use Symfony\Component\Message\Transport\ReceiverInterface;
+use Symfony\Component\Message\Worker;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessageConsumeCommand extends Command
+{
+    protected static $defaultName = 'message:consume';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('receiver', InputArgument::REQUIRED, 'Name of the receiver'),
+                new InputOption('bus', 'b', InputOption::VALUE_REQUIRED, 'Name of the bus to dispatch the messages to', 'message_bus'),
```

Comment:
as a developer I would like to be able:

* Limit number of messages to be processed
* Limit number of time to consume from a queue.
* Exit when particular memory threshold reached. 
* Provide list of queues to consume from, all if none provided.

Reply:
I don't believe these should be part of this pull-request **but** the good news is: it's easy to implement these (first 3 features) agnosticly to the adapter by using `ReceiverInterface` decorators. It would stop `yield`ing when it meets the requirements you described.

The last behaviour would have to be done more low level (i.e. adapter) I'd say, as the idea is to have a receiver that keeps waiting for messages obviously :)

Reply:
I agree with you, we could ship such feature, I'll add that.

Reply:
Added in c98e3d7865.